### DISCUSSION TITLE
Changing hjkl -> jkl;

### DISCUSSION THREAD

**Discussion by lpkha21** [helix-editor/helix#12300] [Q&A] [answered]
In neovim I was able to change hjkl (default moving keymapping) to jkl;

I was able to do the same here in Normal mode:

[keys.normal]
"j" = "move_char_left"           # Left arrow
"k" = "move_visual_line_down"    # Down arrow
"l" = "move_visual_line_up"      # Up arrow
";" = "move_char_right"          # Right arrow

but in visual mode:

[keys.select]
"j" = "move_char_left"           # Extend selection left (same as left arrow)
"k" = "move_visual_line_down"    # Extend selection down (same as down arrow)
"l" = "move_visual_line_up"      # Extend selection up (same as up arrow)
";" = "move_char_right"          # Extend selection right (same as right arrow)
"h" = "collapse_selection"

it doesn't extend select. this only moves the cursor. Is there any way to make this remapping to work on all modes equally.

for example directly mapping the j -> h, k -> j, l -> k, ; -> l

this is the way I solved this issue in Vim.

vim.cmd([[
  nnoremap j h
  nnoremap k j
  nnoremap l k
  nnoremap ; l

  vnoremap j h
  vnoremap k j
  vnoremap l k
  vnoremap ; l
]])

**Comment by kirawi** [comment, marked as answer]
You need to use the `extend_char_XXX` variants

**Reply by lpkha21** [reply]
thanks

**Reply by TornaxO7** [reply]
Don't forget to mark your question as answered if it solved your problem ;)

**Reply by peterjc** [reply]
i.e. Do this

```
[keys.select]
"h" = "collapse_selection" # was on ";" 
"j" = "extend_char_left" # was on "h"
"k" = "extend_visual_line_down" # was on "j"
"l" = "extend_visual_line_up" # was on "k"
";" = "extend_char_right" # was on "l"
```
